tests/unit_tests/config/test_queue_config.py

```diff
@@ -83,22 +83,130 @@ def test_that_invalid_queue_option_raises_validation_error(
     with open(filename, "w", encoding="utf-8") as f:
         f.write("NUM_REALIZATIONS 1\n")
-        f.write("QUEUE_SYSTEM TORQUE\n")
-        f.write(f"QUEUE_OPTION TORQUE MEMORY_PER_JOB {memory_with_unit_str}")
+        f.write(f"REALIZATION_MEMORY {memory_with_unit}\n")
+
+    assert ErtConfig.from_file(filename).queue_config.realization_memory > 0
+
+
+@pytest.mark.parametrize(
+    "memory_spec, expected_bytes",
+    [
+        ("1", 1),
+        ("1b", 1),
+        ("10b", 10),
+        ("10kb", 10 * 1024),
+        ("10mb", 10 * 1024**2),
+        ("10gb", 10 * 1024**3),
+        ("10Mb", 10 * 1024**2),
+        ("10Gb", 10 * 1024**3),
+        ("10Tb", 10 * 1024**4),
+        ("10Pb", 10 * 1024**5),
+    ],
+)
+def test_realization_memory_unit_support(memory_spec: str, expected_bytes):
+    filename = "config.ert"
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write("NUM_REALIZATIONS 1\n")
+        f.write(f"REALIZATION_MEMORY {memory_spec}\n")
+
+    assert (
+        ErtConfig.from_file(filename).queue_config.realization_memory == expected_bytes
+    )
+
+
+@pytest.mark.parametrize("invalid_memory_spec", ["-1", "-1b", "b", "4ub"])
+def test_invalid_realization_memory(invalid_memory_spec: str):
+    filename = "config.ert"
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write("NUM_REALIZATIONS 1\n")
+        f.write(f"REALIZATION_MEMORY {invalid_memory_spec}\n")
+
+    with pytest.raises(ConfigValidationError):
+        ErtConfig.from_file(filename)
+
+
+def test_conflicting_realization_slurm_memory():
+    filename = "config.ert"
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write("NUM_REALIZATIONS 1\n")
+        f.write("REALIZATION_MEMORY 10Mb\n")
+        f.write("QUEUE_SYSTEM SLURM\n")
+        f.write("QUEUE_OPTION SLURM MEMORY 20m\n")
+
+    with pytest.raises(ConfigValidationError), pytest.warns(
+        ConfigWarning, match="deprecated"
+    ):
+        ErtConfig.from_file(filename)
+
+
+def test_conflicting_realization_slurm_memory_per_cpu():
+    filename = "config.ert"
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write("NUM_REALIZATIONS 1\n")
+        f.write("REALIZATION_MEMORY 10Mb\n")
+        f.write("QUEUE_SYSTEM SLURM\n")
+        f.write("QUEUE_OPTION SLURM MEMORY_PER_CPU 20m\n")
+
     with pytest.raises(ConfigValidationError):
         ErtConfig.from_file(filename)
 
 
+def test_conflicting_realization_openpbs_memory_per_job():
+    filename = "config.ert"
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write("NUM_REALIZATIONS 1\n")
+        f.write("REALIZATION_MEMORY 10Mb\n")
+        f.write("QUEUE_SYSTEM TORQUE\n")
+        f.write("QUEUE_OPTION TORQUE MEMORY_PER_JOB 20m\n")
+
+    with pytest.raises(ConfigValidationError), pytest.warns(
+        ConfigWarning, match="deprecated"
+    ):
+        ErtConfig.from_file(filename)
+
+
+def test_conflicting_realization_openpbs_memory_per_job_but_slurm_activated_only_warns():
```

Comment:
Should we warn when using a different queue system?

Reply:
Do you mean to warn when Torque options are specified but Slurm is active? I would say no to that, giving users the opportunity to be flexible with which queue system a config file uses without Ert being too noisy.

Reply:
Kinda. This test tests that using the deprecated torque queue option `MEMORY_PER_JOB` warns even when not running on torque. Is this something we want?

Reply:
ah. 🤔 I think we want to keep this warning, as there is no reason to use MEMORY_PER_JOB even if you don't use Torque. Then you should forward that memory information to your other queue system.